src/Compilers/Core/Portable/ReferenceManager/AssemblyData.cs

```diff
@@ -69,6 +70,17 @@ internal abstract class AssemblyData
             public abstract Compilation? SourceCompilation { get; }
 
             private string GetDebuggerDisplay() => $"{GetType().Name}: [{Identity.GetDisplayName()}]";
+#if DEBUG
```

Comment:
This was in the original fix (Phttps://github.com/dotnet/roslyn/pull/66430). @AlekseyTs?

Reply:
Optimistically removed the code

Reply:
>Is the point sealing the overrides so it can't be overridden further? 

Yes.

>If so, why not do this unconditionally?

Because we don't want to customize the behavior, we just want to prevent others from doing so. Sealing the methods in DEBUG is sufficient to ensure that, and this doesn't cost us anything in RELEASE, including the binary size. At the same time, there is no behavior difference DEBUG vs. RELEASE.